### ISSUE TITLE
Errors while trying to open repository

### ISSUE THREAD

**Issue by v-honcharenko** [ryu1kn/vscode-remote-repo-viewer#2]
Errors while trying to open repository:
```
[Extension Host] TypeError: repository.match is not a function or its return value is not iterable
	at Object.directoryName (/Users/user/.vscode/extensions/ryu1kn.remote-repo-viewer-0.3.0/lib/repository.config.js:49:41)
	at Repository.downloadIfNotExist (/Users/user/.vscode/extensions/ryu1kn.remote-repo-viewer-0.3.0/lib/repository.js:15:37)
	at OpenRepositoryCommand.execute (/Users/user/.vscode/extensions/ryu1kn.remote-repo-viewer-0.3.0/lib/commands/open-repository.js:21:47)
t.log 
[Extension Host] TypeError: Cannot read property 'Symbol(Symbol.iterator)' of null
	at Object.directoryName (/Users/user/.vscode/extensions/ryu1kn.remote-repo-viewer-0.3.0/lib/repository.config.js:49:41)
	at Repository.downloadIfNotExist (/Users/user/.vscode/extensions/ryu1kn.remote-repo-viewer-0.3.0/lib/repository.js:15:37)
	at OpenRepositoryCommand.execute (/Users/user/.vscode/extensions/ryu1kn.remote-repo-viewer-0.3.0/lib/commands/open-repository.js:21:47)
t.log @ 
```

OS: macOS Mojave 10.14.3
VSCode Version: 1.33.1
Plugin Version: 0.3.0

**Comment by ryu1kn** [comment]
Hi @HackedBeat can you provide the steps to reproduce?

**Comment by caironm** [comment]
Cannot read property 'Symbol(Symbol.iterator)' of null

MacOS Mojavi
10.14.4

**Comment by ryu1kn** [comment]
Thanks @caironm , could you let me know the steps to reproduce?

**Comment by v-honcharenko** [comment]
Hi @ryu1kn 
I did the same actions as in the demo:

![image](https://raw.githubusercontent.com/ryu1kn/vscode-remote-repo-viewer/master/images/public.gif)


1) Run command `Open repository in a new window`
2) Paste `https://github.com/ryu1kn/csv-writer.git`
3) Press Enter

**Comment by ryu1kn** [comment]
Thanks @HackedBeat I can reproduce the problem!

**Comment by ryu1kn** [comment]
[This line](https://github.com/ryu1kn/vscode-remote-repo-viewer/blob/9a895116d53e1cabfc9bb05e4b0e9f53ac73e1c5/lib/repository.config.js#L49) is causing the error. If `repository` is `"https://github.com/ryu1kn/csv-writer.git"`, `repository.match(/:([^/]+)\/([^.]+)\.git/)` fails to capture `user` and `repo`. It works if I give `"git@github.com:ryu1kn/csv-writer.git"` and returns `ryu1kn` and `csv-writer`.

**Comment by v-honcharenko** [comment]
@ryu1kn So, here is the solution of this problem:

```
const repoLink = repository.includes('http') 
  ? new URL(repository).pathname 
  : repository
const [,, user, repo] = repoLink.match(/(:|\/)([^/]+)\/([^.]+)\.git/)
```

This code is less elegant than it was before, but it works with two types of links.

**Comment by lukaszmn** [comment]
@HackedBeat the Pull Request's build failed, would you mind fixing it?

@ryu1kn The documented feature to use a shortcut `g ryu1kn/vscode-remote-repo-viewer` also fails.

**Comment by ryu1kn** [comment]
@HackedBeat Thanks for the PR!

**Comment by ryu1kn** [comment]
Released as v0.4.0. Updating ESLint caused a build failure...
